### PR TITLE
cli: Add --no-remote, rename --no-fetch to --cached-only

### LINKED ISSUE #3384: Add --no-remote to disable remote module resolution
The behaviour of `--no-fetch` is to fail to resolve remote modules unless they are already in the cache.

A ~`--no-remotes`~`--no-remote` flag that simply disabled remote module resolution would be more useful and reasonable while not conflicting with Deno's philosophy any more than `--no-fetch` does already.

### PR DESCRIPTION
**Pull request by nayeemrmn**
Closes #3384.

### CODE REVIEW

File: cli/file_fetcher.rs [hunk 1]
```diff
@@ -196,13 +197,17 @@ impl SourceFileFetcher {
...
+  /// If `no_fetch` is true then if remote file is not found in disk
   /// cache this method will fail.
+
   fn get_source_file_async(
     self: &Self,
     module_url: &Url,
     use_disk_cache: bool,
-    no_remote_fetch: bool,
+    no_remote: bool,
+    no_fetch: bool,
```
Comment (nayeemrmn) [inline cli/file_fetcher.rs]: This argument and the functionality of `--no-fetch` is still required internally for this bit: https://github.com/denoland/deno/blob/07e78f64bbda01e6a66ee4b948e4ac5e0e86cb53/cli/file_fetcher.rs#L124-L129

Comment (bartlomieju) [reply cli/file_fetcher.rs]: @nayeemrmn I don't understand why... Isn't it just renaming of the flag?

Comment (nayeemrmn) [reply cli/file_fetcher.rs]: Read #3384 for details of the difference in the flags. I'm saying here that the (no longer exposed) functionality of `--no-fetch` is still required for the above function, so we need both arguments here.

Comment (hayd) [reply cli/file_fetcher.rs]: This is a very confusing change.

**Comment by hayd** [comment]
This PR replaces/removes --no-fetch, to match #3384 change it should only add the flag.

As mentioned in the other issue I think `--no-fetch` is very useful. Another motivating example is to have `deno fetch deps.ts` then (at build time) `deno fetch --no-fetch main.ts` and (at runtime) `deno run --no-fetch main.ts` i.e. force all urls to be defined in `deps.ts`. This is useful for fast iterating in docker e.g. [dockerfile with multi fetches](https://github.com/hayd/deno-lambda/blob/7b8057607cc35cc6390c01b297ca3d75e3d3f8dc/tests/Dockerfile#L26).

**Comment by nayeemrmn** [comment]
@hayd Yeah, I just wanted to ask first if the added flag still sounded good. I guess there's some interest.

@ry Please review.

**Comment by ry** [comment]
@nayeemrmn There's something about this that strikes me as confusing or inelegant. 
```
> deno help run
[...]
        --no-fetch                     Do not download remote modules
        --no-remote                    Do not resolve remote modules

> deno help fetch
[...]
        --no-remote                   Do not resolve remote modules
```
I acknowledge there is a use-case for both of these flags, but I can't help but wonder if they could be merged into a single concept.

I think it would be very helpful to users if a section in the manual was added (perhaps to https://deno.land/std/manual.md#linking-to-third-party-code ) which explained the difference and use cases for these flags in more detail...

**Comment by ry** [comment]
The documentation should describe these situations:
```
deno run --no-remote
deno run --no-fetch
deno run --no-remote --no-fetch
deno fetch --no-remote
``` 

It seems you would never need `deno run --no-remote --no-fetch`, because `--no-remote` is strictly stronger than `--no-fetch`. That suggests what perhaps we want a single flag that describes the behavior for remote modules...

**Comment by hayd** [comment]
What about a flag with a value? e.g. --no-remote=download or --no-remote=use.
(likely there is something semantically more pleasing...)

**Comment by nayeemrmn** [comment]
The purpose of `--no-fetch` is to fail at `run` if any dependencies are missing from the cache. A security feature for a CI edge case. `--no-remote` just bans remote modules.

Correct me if I'm wrong, but the similarity between `--no-fetch` and `--no-remote` is an illusion caused by `--no-fetch` weirdly using module resolution errors and maybe its misleading name. They actually have nothing to do with each other and can co-exist just fine with some changes:

My suggestion is to rename `--no-fetch` to `--cached-only`, give it its own error
(`--cached-only was specified but dependency x is not cached`) and an appropriate help message
(`Fail if any remote dependencies are not cached in advance`). That makes the distinction a lot clearer.

Yes, `--no-remote` would render `--cached-only` meaningless but that would only be intuitive.

**Comment by nayeemrmn** [comment]
@ry I've implemented my above suggestion. Please review.

**Pull request merged**